src/utils/misc/wait.ts

```diff
@@ -1,11 +1,11 @@
 import {type Instance} from '../../setup'
 
-export function wait(config: Instance['config']) {
+export async function wait(config: Instance['config']): Promise<void> {
   const delay = config.delay
   if (typeof delay !== 'number') {
     return
   }
-  return Promise.all([
+  await Promise.all([
```

Comment:
This makes the function return `Promise<undefined>` instead of `undefined` for `delay: null` and results in `await wait()` to add a new micro task.
Are we sure this has no effect?

Reply:
Doesn't `await` always create a new microtask regardless? Doesn't matter what the right-hand-side is, right?

Reply:
I'm not sure. I guess it depends on implementation.

Typescript reports `'await' has no effect on the type of this expression. ts(80007)` if you try `await undefined`.

The following results in `sync` in `node@10` and `async` in `node@12`:
```js
async function waitAsync() {
    return undefined
}

function waitSync() {
    return undefined
}

(async function () {
    await Promise.race([
        (async () => {
            await waitAsync()
            return 'async'
        })(),
        (async () => {
            await waitSync()
            return 'sync'
        })(),
    ]).then(console.log)
})()
```

If we're not sure that this has no effect on any platform that is still supported, I wouldn't touch it if I don't have to.